### PR TITLE
Add ability to export a dataset into html formatted with conditional formatting

### PR DESCRIPTION
**Pull request by apichery**
Need https://github.com/dataiku/dip/pull/28655 for it to work

### CODE REVIEW

**Comment by thomaswartel** [comment]
@apichery it works very well, thanks a lot!

I think we should update the recipe.json to update the condition when to display the conditional formatting. Now, conditional formatting is no longer related to Excel files only. One can apply a conditional format in the body + Excel independently 👍 

https://github.com/dataiku/dss-plugin-sendmail/assets/105646698/45068ec3-7df2-46f4-ab3d-43404eaaea48


`{
            "name": "apply_coloring_excel",
            "label" : "Apply conditional formatting",
            "description" : "Color cells by rules, when applicable in the HTML body. Will also be applied when attachment format is Excel.",
            "type": "BOOLEAN",
            "visibilityCondition" : "(model.body_format == 'html') || (model.attachment_type == 'excel_can_ac')"
        }`

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,13 +1,18 @@
 # Changelog
 
+## [Version 1.0.1](https://github.com/dataiku/dss-plugin-sendmail/releases/tag/v1.0.1) - Feature release - 2024-04
```
Comment (liamlynch-data) [inline CHANGELOG.md:3]: We need to up the version number in the plugin json also

**Review by liamlynch-data — APPROVED** [review]
Looks fine - just needed to bump the version in the JSON.
Also will implement Thomas's suggestion in another commit

File: CHANGELOG.md (hunk 1 above)
Comment (liamlynch-data) [reply CHANGELOG.md:3]: Done here https://github.com/dataiku/dss-plugin-sendmail/pull/12/commits/2ed8532b938a4cd078ee5812a3a74d678450db9b

**Review by liamlynch-data — CHANGES_REQUESTED** [review]
Actually not sure if the param to toggle the formating is being used (seems to be always on). I will fix it

**Comment by liamlynch-data** [comment]
Here https://github.com/dataiku/dss-plugin-sendmail/pull/12/commits/7ac69c80ca70bb7df8d44a6d49c17d2a0de8d42b I
- implemented the suggestion of @thomaswartel 
- made the parameter control whether formating applied, for HTML as well as excel

(Another choice would be that the formatting is always applied for the HTML output and the toggle only applies for excel -if that's what we really want, revert this commit as that was how it behaved before)

Note - I also tested on old DSS version (12.5.2), and range of python versions (3.6, 3.9, 3.11)

**Review by liamlynch-data — APPROVED** [review]

**Comment by liamlynch-data** [comment]
- Added send_no_attachments option  and changed defaults behaviour for attachments 
- Added some more section headings
- Conditional formating on by default (if excel attached or HTML output used)

As agreed with @thomaswartel  and @MayeulRousselet

File: python-lib/dku_attachment_handling.py [hunk 2]
```diff
@@ -14,16 +15,20 @@ def attachments_template_dict(attachment_datasets, home_project_key):
...
             attachment_entry = attachments_dict.setdefault(ds_name, {})
         else:
-            # For foreign datasets, we need another level in the map with the
+            # For foreign datasets, we need another level in the map with the project key
             ext_project_entry = attachments_dict.setdefault(attachment_ds.project_key, {})
             attachment_entry = ext_project_entry.setdefault(ds_name, {})
 
-        attachment_entry["html_table"] = table_df.to_html(index=False, justify='left', border=0, na_rep="")
+        # Use DSS to_html method if available (DSS 12.6.2+)
+        if hasattr(attachment_ds.__class__, "to_html") and callable(getattr(attachment_ds.__class__, "to_html")):
```
Comment (MayeulRousselet) [inline python-lib/dku_attachment_handling.py]: Nitpick: Extract this check in a method `supports_dataset_to_html`

**Review by MayeulRousselet — APPROVED** [review]
LGTM!

**Comment by liamlynch-data** [comment]
I will address the nitpick of @MayeulRousselet then merge this PR - then move on to the [subsequent ticket](https://app.shortcut.com/dataiku/story/180121/plugin-migration-update-issue-existing-recipes-on-dss-12-6-with-excel-as-an-attachement-will-show-nothing-selecting-when) - unless objections

File: python-lib/dku_attachment_handling.py (hunk 2 above)
Comment (liamlynch-data) [reply python-lib/dku_attachment_handling.py]: Addressed [here](https://github.com/dataiku/dss-plugin-sendmail/pull/12/commits/148ad97f37abbf20283483efab3dfaca8d1e5a72)

**Pull request merged**